src/iterators.rs

```diff
@@ -905,6 +906,57 @@ custom_vec_iter_impl!(
 );
 impl PyGCProtocol for Chains {}
 
+impl PyHash for NodeIndices {
+    fn hash<H: Hasher>(&self, py: Python, state: &mut H) -> PyResult<()> {
+        PyHash::hash(&self.nodes, py, state)?;
+        Ok(())
+    }
+}
```

Comment:
If this is not requried I'd rather not make this type hashable. We reserve the right to make it mutable one day which would make it not hashable

Reply:
I implemented this for the `RelationalCoarsestPartition` return type, since `custom_vec_iter_impl!` requires this trait. If you would prefer, I can make a new iterator that will act as a container. 

Since `RelationalCoarsestPartition` is also a pretty specific type name i could change it to something like `NodePartition` containing something such as `NodePartitionBlock` or such.

Reply:
In this case I would implement a new type and add a macro `custom_vec_iter_impl!` calling the old `custom_vec_iter!` and adding the PyHash. The catch is that sometimes these methods can have unintended side effects on performance (e.g. #1096) so it is better to be conservative

Reply:
I am not sure I understood this but the hashing from NodeIndices was removed